src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -99,5 +102,50 @@ public LambdaComparer(Func<T?, T?, bool> equal, Func<T?, int> getHashCode)
 
             public int GetHashCode(T obj) => _getHashCode(obj);
         }
+
+        private const string RegexGeneratorAttributeSource =
+@"#nullable enable
+using global::System.Diagnostics;
+using global::System.Diagnostics.CodeAnalysis;
+using global::System.Threading;
+ 
+namespace System.Text.RegularExpressions
+{ 
+    /// <summary>Instructs the System.Text.RegularExpressions source generator to generate an implementation of the specified regular expression.</summary>
+    /// <remarks>The generator associated with this attribute only supports C#.  It only supplies an implementation when applied to static, partial, parameterless, non-generic methods that are typed to return <see cref=""Regex""/>.</remarks>
+    [Conditional(""INCLUDE_SYSTEM_TEXT_REGULAREXPRESSIONS_REGEXGENERATORATTRIBUTE"")]
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    internal sealed class RegexGeneratorAttribute : Attribute
```

Comment:
I suppose this is a completely separate discussion, so this PR shouldn't block by it, but what are our thoughts around the documentation of the APIs that ship as part of an analyzer? For example, have we thought how to ensure the APIs are documented in docs.microsoft.com? cc@calope